src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -515,8 +524,10 @@ public Void caseTemplatizedType(TemplatizedType type) {
           }
           emit(templateTypeName);
           emit("<");
-          for (JSType tmplType : type.getTemplateTypes()) {
-            visitType(tmplType);
+          UnmodifiableIterator<JSType> it = type.getTemplateTypes().iterator();
+          while (it.hasNext()) {
+            visitType(it.next());
+            if (it.hasNext()) emit(",");
```

Comment:
why not use visitTemplateTypes? I feel we have the same logic in two places, because you already fixed adding ',' for multiple template params.